code/ship/ship.cpp

```diff
@@ -18985,3 +18986,48 @@ ship_subsys* ship_get_subsys_for_submodel(ship* shipp, int submodel)
 
 	return nullptr;
 }
+
+/**
+ * Is the requested type of ship arriving in the requested stage?
+ *
+ * Dock leaders & followers need to be handled separately such that
+ * leaders handle the warp process, but followers need to behave as though
+ * warping with respect to targeting, drawing thrusters, radar, AI
+ * targeting, etc
+ *
+ * @param stage Check one of stage1, stage2 or both
+ * @param dock_leader_only only the dock leader, or leader/followers/non-docked?
+ *
+ * @return is the ship arriving, bool
+ */
+bool ship::is_arriving(ship::warpstage stage, bool dock_leader_only)
```

Comment:
This looks a lot better; I just didn't realize it would be so complicated!

Reply:
There remains one small area of confusion.  I thought the division would be dock leaders and single arriving ships on the one side, and dock followers on the other.  But with your boolean variable, it's dock leaders on one side, and everything else on the other.  That doesn't match the logic of the code, unless I'm missing something.

Reply:
It's also worth considering making the second parameter an enum, even if there are only two options.

Reply:
aw dang; you're correct, I got myself confused! I'll rename the param, switch the calling statements if required & update the doco. 

Regarding enum vs bool; as it's only a binary affair (and I can't see how it'd be extended to a 3rd state) I'd prefer to keep the bool

Reply:
Okey dokey.